bindings/matrix-sdk-ffi/src/error.rs

```diff
@@ -6,57 +8,63 @@ pub enum ClientError {
     Generic { msg: String },
 }
 
+impl ClientError {
+    fn new<E: Display>(error: E) -> Self {
+        Self::Generic { msg: error.to_string() }
```

Comment:
Should we even loop over the `source()`s and try to print the error as well as all its underlying causes?